### ISSUE TITLE
TEP0107 & TEP0075 - Issues with propagating object parameters

### ISSUE THREAD

**Issue by chuangw6** [tektoncd/pipeline#5140]
In TEP0107's implementation #4845 , `Parameters` are propagated directly from taskrun/pipelinerun to the variable replacements in embedded specifications without  having to define param spec in `TaskSpec.Params[*]`.

In TEP0075's implementation, we asks users to provide the object schema (keys) in `TaskSpec.Params[*].Properties` section as per [the TEP0075 proposal](https://github.com/tektoncd/community/blob/main/teps/0075-object-param-and-result-types.md#adding-additional-json-schema-properties).

In short, object params have to define param specs but in param propagation, param specs are missed. Talked to @chitrangpatel about this offline. We landed on an idea that parameter propagation might not be applicable for object params. 

If anyone has any comments/ideas/concerns, please feel free to add them here.

**Comment by chuangw6** [comment]
cc @jerop @dibyom @lbernick @vdemeester

**Comment by vdemeester** [comment]
> In short, object params have to define param specs but in param propagation, param specs are missed. Talked to @chitrangpatel about this offline. We landed on an idea that parameter propagation might not be applicable for object params.

Having parameter propagation working *depending* on the type of parameters doesn't make too much sense to me. Either propagation works for all or none. We need to find a way for it to work for object params as well 🤞🏼

**Comment by jerop** [comment]
fixed in https://github.com/tektoncd/pipeline/pull/5371

**Comment by jerop** [comment]
Thank you @chitrangpatel!
